velox/functions/lib/aggregates/MinMaxByAggregatesBase.h

```diff
@@ -497,10 +500,16 @@ class MinMaxByAggregateBase : public exec::Aggregate {
   inline void updateValues(
       char* group,
       const DecodedVector& decodedValues,
-      const DecodedVector& decodedComparisons,
+      DecodedVector& decodedComparisons,
       vector_size_t index,
       bool isValueNull,
       MayUpdate mayUpdate) {
+    if (throwOnNestedNulls_) {
+      const auto* indices = decodedComparisons.indices();
+      velox::aggregate::prestosql::checkNestedNulls(
```

Comment:
checkNestedNulls takes a const reference? Why we change it a reference?

Reply:
DecodedVector::indices is a method with side effects, we could not pass a const reference if we want to use this method here.
```C++
const vector_size_t* indices() {
    if (!indices_) {
      fillInIndices();
    }
    return &indices_[0];
  }
```